cmd/certificate_maker/certificate_maker.go

```diff
@@ -0,0 +1,182 @@
+	createCmd.Flags().StringVar(&intermediateKeyID, "intermediate-key-id", "", "KMS key identifier for intermediate certificate")
+	createCmd.Flags().StringVar(&intermediateTemplate, "intermediate-template", "pkg/certmaker/templates/intermediate-template.json", "Path to intermediate certificate template")
+	createCmd.Flags().StringVar(&intermediateCert, "intermediate-cert", "intermediate.pem", "Output path for intermediate certificate")
+	createCmd.Flags().StringVar(&kmsVaultToken, "vault-token", "", "HashiVault token")
+	createCmd.Flags().StringVar(&kmsVaultAddr, "vault-address", "", "HashiVault server address")
+}
+
+func runCreate(_ *cobra.Command, _ []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Build KMS config from flags and environment
+	config := certmaker.KMSConfig{
+		Type:              getConfigValue(kmsType, "KMS_TYPE"),
+		Region:            getConfigValue(kmsRegion, "AWS_REGION"),
+		RootKeyID:         getConfigValue(rootKeyID, "KMS_ROOT_KEY_ID"),
+		IntermediateKeyID: getConfigValue(intermediateKeyID, "KMS_INTERMEDIATE_KEY_ID"),
+		LeafKeyID:         getConfigValue(leafKeyID, "KMS_LEAF_KEY_ID"),
+		Options:           make(map[string]string),
+	}
+
+	// Handle KMS provider options
+	switch config.Type {
+	case "gcpkms":
+		if credsFile := getConfigValue(kmsCredsFile, "GCP_CREDENTIALS_FILE"); credsFile != "" {
+			// Check if credentials file exists before trying to use it
+			if _, err := os.Stat(credsFile); err != nil {
+				if os.IsNotExist(err) {
+					return fmt.Errorf("failed to initialize KMS: credentials file not found: %s", credsFile)
+				}
+				return fmt.Errorf("failed to initialize KMS: error accessing credentials file: %w", err)
+			}
+			config.Options["credentials-file"] = credsFile
+		}
+	case "azurekms":
+		if tenantID := getConfigValue(kmsTenantID, "AZURE_TENANT_ID"); tenantID != "" {
+			config.Options["tenant-id"] = tenantID
+		}
+	case "hashivault":
+		if token := getConfigValue(kmsVaultToken, "VAULT_TOKEN"); token != "" {
+			config.Options["token"] = token
+		}
+		if addr := getConfigValue(kmsVaultAddr, "VAULT_ADDR"); addr != "" {
+			config.Options["address"] = addr
+		}
+	}
+
+	km, err := certmaker.InitKMS(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to initialize KMS: %w", err)
+	}
+
+	// Validate template paths
+	if err := certmaker.ValidateTemplatePath(rootTemplatePath); err != nil {
+		return fmt.Errorf("root template error: %w", err)
+	}
+	if err := certmaker.ValidateTemplatePath(leafTemplatePath); err != nil {
+		return fmt.Errorf("leaf template error: %w", err)
+	}
+
+	return certmaker.CreateCertificates(km, config, rootTemplatePath, leafTemplatePath, rootCertPath, leafCertPath, intermediateKeyID, intermediateTemplate, intermediateCert)
+}
+
+func main() {
+	if err := rootCmd.Execute(); err != nil {
+		logger.Fatal("Command failed", zap.Error(err))
+	}
+}
+
+func getConfigValue(flagValue, envVar string) string {
```

Comment:
as an aside, i've used https://github.com/spf13/viper?tab=readme-ov-file#working-with-flags to bind flags and env vars so that you can just call `viper.GetString()` on a flag and it reconciles the sources in a consistent order.

we're already using this as a dependency within fulcio so perhaps it is less footprint to reuse that?

Reply:
oh nice, that's a lot cleaner / thanks.